control_panel_api/authentication.py

```diff
@@ -0,0 +1,47 @@
+from django.conf import settings
+import jwt
+from jwt.exceptions import InvalidTokenError
+from rest_framework.authentication import (
+    BaseAuthentication,
+    get_authorization_header
+)
+from rest_framework.exceptions import AuthenticationFailed
+
+from control_panel_api.models import User
+
+
+class Auth0JWTAuthentication(BaseAuthentication):
+
+    def authenticate(self, request):
+
+        try:
+            prefix, token = get_authorization_header(request).split()
+
+        except (ValueError, TypeError):
+            return None
+
+        if prefix != 'JWT'.encode():
+            raise AuthenticationFailed('JWT prefix missing')
+
+        try:
+            decoded = jwt.decode(
+                token,
+                key=settings.AUTH0_CLIENT_SECRET,
```

Comment:
This might be quite an annoying suggestion, but would it be better to call these vars `OIDC_CLIENT_*` rather than referencing a specific vendor?